exercises/sieve/src/sieve.h

```diff
@@ -1,10 +1,13 @@
 #ifndef SIEVE_H
 #define SIEVE_H
 
-#define MAX_LIMIT_TESTED (1000)
-
-typedef unsigned int primes_array_t[MAX_LIMIT_TESTED];
-
-unsigned int sieve(const unsigned int limit, primes_array_t primes);
+#include <stdint.h>
+#include <stddef.h>
+
+/// Calculate at most `max_primes` prime numbers in the interval [2,limit] 
+/// using the Sieve of Eratosthenes and store the prime numbers in `primes`
```

Comment:
Should not mention implementation requirements here, this should be covered by the readme (or additional hints to the readme)

Reply:
The exercise is called *sieve* and the README explicitly states that students should use the algorithm. But the majority of published solutions I've seen do not use the Sieve of E. at all. Given that the function is also named `sieve`, I wouldn't consider the requirement of using the Sieve of E. an implementation detail.

Reply:
Yes it is covered by the readme. The readme is populated automatically from the exercism/problem-specifications repo. We should not duplicate the information in the exercise where it is not in synch with that repo. If students are not reading the readme, that is a separate issue (that should also be dealt with but is not specific to this exercise)

Reply:
@wolf99 I see your point, but we're unlikely to contradict the README in this case. I suspect there will still be plenty of non-sieve solutions, but this seems like reasonable documentation for the function itself.